graph/oracle-compare.go

```diff
@@ -71,8 +71,12 @@ func CompareToOracle[V VPI[V], E EPI[E], M MVI[M], N any, A Algorithm[V, E, M, N
 
 		log.Info().Msg("Creating result for graph with " + utils.V(numVertices) + " vertices and " + utils.V(numEdges) + " edges")
 		oracleGraph.AlgTimer.Start()
-		ConvergeAsync(alg, oracleGraph, new(sync.WaitGroup))
-		if aOF, ok := any(alg).(AlgorithmOnFinish[V, E, M, N]); ok {
+		algOracle := alg
+		if aN, ok := any(alg).(AlgorithmNew[V, E, M, N, A]); ok {
+			algOracle = aN.New()
```

Comment:
Shouldn't this also be done in the main flow as well as here in the oracle section?

Reply:
Yeah you're right. I was assuming the user would have called New() themselves before launching the algorithm.